lib/routes/account.js

```diff
@@ -1914,8 +1995,162 @@ module.exports = function (
             },
+            function (err) {
+              reply(err)
+            }
+          )
+      }
+    },
+    {
+      method: 'GET',
+      path: '/recovery_email/emails',
+      config: {
+        auth: {
+          strategy: 'sessionTokenWithVerificationStatus'
+        },
+        response: {
+          schema: isA.array().items(
+            isA.object({
+              isVerified: isA.boolean().required(),
+              isPrimary: isA.boolean().required(),
+              email: validators.email().required()
+            }))
+        }
+      },
+      handler: function (request, reply) {
+        const sessionToken = request.auth.credentials
+        const uidHex = sessionToken.uid.toString('hex')
+        const userEmail = sessionToken.email
+        var emails
+
+        log.begin('Account.RecoveryEmailEmail', request)
+
+        customs.check(request, userEmail, 'accountEmails')
+          .then(getAccountEmails)
+          .then(createResponse)
+          .done(reply, reply)
+
+        function getAccountEmails() {
+          return db.accountEmails(uidHex)
+            .then(function (result) {
+              emails = result
+            })
+        }
+
+        function createResponse() {
+          return emails.map((email) => {
+            return {
+              email: email.email,
+              isPrimary: !! email.isPrimary,
+              isVerified: !! email.isVerified
+            }
+          })
+        }
+      }
+    },
+    {
+      method: 'POST',
+      path: '/recovery_email/create',
+      config: {
+        auth: {
+          strategy: 'sessionTokenWithVerificationStatus'
+        },
+        validate: {
+          payload: {
+            email: validators.email().required()
+          }
+        },
+        response: {}
+      },
+      handler: function (request, reply) {
+        const sessionToken = request.auth.credentials
+        const uidHex = sessionToken.uid.toString('hex')
+        const userEmail = sessionToken.email
+        const ip = request.app.clientAddress
+        const email = request.payload.email
+        var emailCode
+
+        log.begin('Account.RecoveryEmailCreate', request)
+
+        customs.check(request, userEmail, 'createEmail')
+          .then(generateRandomValues)
+          .then(createEmail)
+          .then(sendEmailVerification)
+          .then(
+            function () {
+              reply({})
+            },
             reply
           )
+
+        function generateRandomValues() {
+          return random(16)
+            .then(bytes => {
+              emailCode = bytes
+            })
+        }
+
+        function createEmail() {
+          var emailData = {
+            email: email,
+            normalizedEmail: email.toLowerCase(),
+            emailCode: emailCode,
+            isVerified: false,
+            isPrimary: false,
+            uid: uidHex,
+            createdAt: Date.now()
+          }
+          return db.createEmail(uidHex, emailData)
```

Comment:
Many of the other `db.*` methods appear to convert to hex internally, should we pass in the raw bytes here for consistency?